### DISCUSSION TITLE
hash must be unicode or bytes, not sqlalchemy.orm.attributes.InstrumentedAttribute

### DISCUSSION THREAD

**Discussion by varunponda** [fastapi/fastapi#6310] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
app=FastAPI()

@app.post('/user')
def createUser():
    userPassword=user(password:hashPass(userModel.password))

def verifyPass(plain_password,hashed_password):
        pwd_context=CryptContext(schemes=["bcrypt"],deprecated="auto")
        verified=pwd_context.verify(plain_password,hashed_password)
        return verified

@app.post('/login')
def verifyUser():
    check=verifyPass(pass,user.password)
    return 'user verified'
```

### Description

I have created a user which has hashed password using pwd_context.hash(). I want to verify whether the hash password and the plain password(from user login) are same or not with pwd_context.verify(). but I get the error:
hash must be unicode or bytes, not sqlalchemy.orm.attributes.InstrumentedAttribute

### Operating System

Windows

### FastAPI Version

0.80.0

### Python Version

3.10.6

**Comment by iudeen** [comment]
hmm, how is this a FastAPI question/issue?

**Comment by rati90** [comment, marked as answer]
Check the docs https://fastapi.tiangolo.com/advanced/security/oauth2-scopes/ < Here is a pretty good explanation.

**Comment by tiangolo** [comment]
Thanks for the help here everyone! 👏  🙇 

Thanks for coming back and closing the issue @varunponda  👍
